Topics/python-fundamentals.md

```diff
@@ -0,0 +1,236 @@
+
+_Any existing software project that is not under version control is considered a dead project, and the responsible developers are considered crazy._
+
+Data Science projects (which make heavy use of software) are no different, indeed! 
+
+They also have the _additional problem of data versioning_, which is the raw material on which you work most. 
+
+[**Here**](https://rogerdudler.github.io/git-guide/index.html) you can find a simple guide to Git.
+
+**Learn it, it's freaking worth (and necessary).**
+
+Documenting well your work with Git is crucial: read [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+
+**Must read:**
+[**Ten Simple Rules for Taking Advantage of Git and GitHub**](http://dx.plos.org/10.1371/journal.pcbi.1004947)
+
+### Ask Questions
+A rule of thumb to learn fast and effectively is to ask questions and read other's questions and answers.
+
+Join communities of people interested in the topic (e.g. Reddit): here you can find discussions, search by keywords (e.g. "matrix multiplication"), and ask questions, with experts who will answer and help you.
+
+Some _tips_ regarding questions:
+
+- Try to form specific, well-written questions, to minimize the time used by the respondent.
+- Do not ask a question whose answer is found with a quick search on google.
+- If the questions are too general or show laziness they'll likely remain unanswered...
+
+Some subreddits you can subscribe to are:
+
+- [r/Python](https://www.reddit.com/r/python/)
+- [r/LearnPython](https://www.reddit.com/r/learnpython/)
+- [r/LearnProgramming](https://www.reddit.com/r/learnprogramming/)
+- [r/Programming](https://www.reddit.com/r/programming/)
+- [r/Computer Science](https://www.reddit.com/r/compsci/)
+
+Other good places to post and read (well structured) questions are:
+
+- [HackerNews](http://hn.premii.com/)
+- [Quora](https://www.quora.com/)
+- [Math Stack Exchange](https://maths.stackexchange.com)
+
+### Conclusions
+
+The path to becoming a good programmer is long and requires commitment and dedication, but it will give you satisfaction (and value) like very few other things in life!
+
+Programming is creativity, problem-solving, art! 
+
+If you follow the advice of this guide you won't have any problem to become proficient with Python, and moreover, you will have learned the main concepts of programming:
+
+This means that to learn another language, the biggest obstacle will be to learn the syntax, but the concepts will remain more or less the same!
+
+In the next guide we will see how to use the Jupyter Notebooks, a Python application designed specifically for Data Science and experimental programming!
+
```

Comment:
I suggest adding [this link](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/Programming-in-Python.md#basics--learning) somewhere in the guide, or towards the end.

Reply:
ok